### PR TITLE
[Merged by Bors] - Fix flaky app test

### PR DESCRIPTION
**Pull request by antonlerner**
Fixed closing sync and block listener

### CODE REVIEW

File: sync/syncer.go [hunk 1]
```diff
@@ -215,12 +215,13 @@ func (s *Syncer) Close() {
 	close(s.exit)
 	close(s.forceSync)
 	s.peers.Close()
+	s.syncLock.Lock()
+	s.syncLock.Unlock()
```
Comment (y0sher) [inline sync/syncer.go:220]: why not close inside the lock? this could be still be a bit racy

**Review by y0sher — APPROVED** [review]

File: sync/syncer.go (hunk 1 above)
Comment (antonlerner) [reply sync/syncer.go:220]: the lock is taken to make sure close has been processed and the loop has exited. after this happens there can be not more requests to sync

**Comment by antonlerner** [comment]
bors merge

**Comment by antonlerner** [comment]
bors merge

**Pull request closed without merging**